core/templates/dev/head/components/ck-editor-helpers/ck-editor-5-rte.directive.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for CK Editor 5.
+ */
+
+require('services/ContextService.ts');
+require('services/RteHelperService.ts');
+
+const ClassicEditor = require(
+  '@ckeditor/ckeditor5-build-classic/build/ckeditor.js');
+
+var oppia = require('AppInit.ts').module;
+
+oppia.directive('ckEditor5Rte', [
+  'ContextService', 'RteHelperService', 'PAGE_CONTEXT',
+  function(ContextService, RteHelperService, PAGE_CONTEXT) {
+    return {
+      restrict: 'E',
+      scope: {
+        uiConfig: '&'
+      },
+      template: '<div><div></div>' +
+                '<div contenteditable="true" class="oppia-rte">' +
+                '</div></div>',
+      require: '?ngModel',
+
+      link: function(scope: ICustomScope, el, attr, ngModel) {
+        var _RICH_TEXT_COMPONENTS = RteHelperService.getRichTextComponents();
+        var names = [];
+        var icons = [];
+        var canUseFs = ContextService.getPageContext() ===
```

Comment:
@aks681 has made some changes such that multiple pages with RTEs can use the file system (e.g. topics etc.). You should probably consult with him on this in case further changes are need to either your or his PRs.

Reply:
Hi @seanlip, I'll read more about it and update you here.
The changes made by @aks681 in fs seems to relatable here I'm looking into it

Reply:
Sounds good. Consider actually talking to @aks681 too, early in the process -- he can give you feedback on your plan.

Reply:
I learned for @aks681 PR for new FS model and I've implemented the same here.
@aks681 PTAL Thanks!